### PR TITLE
Add support and documentation for R: prefix

### PR DESCRIPTION
**Pull request by Elijas**
Original thread:
https://github.com/MobileDynasty/pytest-env/pull/12

### CODE REVIEW

File: src/pytest_env/plugin.py [hunk 1]
```diff
@@ -23,9 +23,21 @@ def pytest_load_initial_conftests(
         key = part[0].strip()
         value = part[2].strip()
 
+        # use R: as a way to designate whether to use 
+        # "raw" value (skip replacing environment
+        # variables in a value). Use this to allow
+        # curly bracket characters in a value.
+        rkey = key.split("R:")
```
Comment (Elijas) [inline src/pytest_env/plugin.py:30]: Using the same code conventions and patterns as the rest of the file.

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Needs test.

**Comment by Elijas** [comment]
@gaborbernat Try now.

**Comment by gaborbernat** [comment]
Can you please rebase on main?

File: tests/example.py [hunk 2]
```diff
@@ -3,5 +3,36 @@
 import os
 
 
-def test_works() -> None:
-    assert "MAGIC" not in os.environ
+class TestSimpleBehavior:
```
Comment (gaborbernat) [inline tests/example.py:6]: please no classes, methods are enough

File: tests/example_pytest.ini [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[pytest]
```
Comment (gaborbernat) [inline tests/example_pytest.ini:1]: Let's not conflate multiple tests into one, and please move the conf closer to the test as shown in the current test example.

File: src/pytest_env/plugin.py [hunk 4]
```diff
@@ -6,9 +6,14 @@
 import pytest
 
 
+_DEFAULT_FLAG = 'D'
+_RAW_FLAG = 'R'
+
+_ALLOWED_FLAGS = (_DEFAULT_FLAG, _RAW_FLAG)
```
Comment (gaborbernat) [inline src/pytest_env/plugin.py:12]: Add final type annotations to these

**Comment by gaborbernat** [comment]
Please actually rebase.

**Comment by Elijas** [comment]
Development moved to #8

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -48,6 +48,6 @@ repos:
         - flake8-comprehensions==3.10
         - flake8-pytest-style==1.6
         - flake8-spellcheck==0.28
-        - flake8-unused-arguments==0.0.11
+        - flake8-unused-arguments==0.0.12
```
Comment (gaborbernat) [inline .pre-commit-config.yaml:51]: Why do these changes show up here, did you rebase?

File: src/pytest_env/plugin.py [hunk 6]
```diff
@@ -2,40 +2,57 @@
 from __future__ import annotations
 
 import os
+from typing import Final
 
 import pytest
 
+_DEFAULT_FLAG: Final = 'D'
```
Comment (gaborbernat) [inline src/pytest_env/plugin.py:9]: Needs type arg in []

**Pull request closed without merging**